### ISSUE TITLE
Weather Dataset prediction always FALSE

### ISSUE THREAD

**Issue by flipdazed** [flipdazed/weather-modelling#1]
**Issue**
The extension to the weather data is a boolean classification task. The affected models are both only returning `False` for all entries.

**Affected**

    train_mlp.py
    train_db.py

**Reproduction**

    ./predict.py -m dump/models/result_dbn.pkl -s 0,-1,10
    ./predict.py -m dump/models/result_mlp.pkl -s 0,-1,10

**Runlog** _(open in OS X terminal with `cat runlog.txt` to enable full colour output)_
 - [DBN runlog](https://github.com/flipdazed/weather-modelling/files/520127/runlog.txt)

**Comment by flipdazed** [comment]
Is the cause bad parameters or related to the cost issue of #2

**Comment by flipdazed** [comment]
**nodes**
Here I tried `[15000, 15000, 15000]`. I assumed I would need `>10000` nodes in at least the first RBM layer to model cross-relationships between the data sets. I made this assumption from reading [this](http://www.cs.toronto.edu/~rsalakhu/papers/science.pdf) paper by Geoffrey Hinton who seemed to arbitrarily just bump up the first number of nodes depending on the complexity of the task at hand. I’m not entirely sure how complex this particular task is though and how many nodes to include and how to distribute them.

**learning rate, epochs, batch size, early stopping**
I don’t think this is the problem here as these would just cause the learning to be slower/faster rather than the wrong direction. Mini-batch size seems reasonable at 40 for a total sample size of 582. Early stopping is not part of the pre-training so can’t be responsible for the direction issues

**`k`, number of Gibbs steps in CD/PCD**
Is this possibly responsible? `1` is the lowest this could possibly be!

**Comment by flipdazed** [comment]
# Training of Weights
A sample of the weights showing activity early on then nothing for the rest of the run time
![train_mlp_weights](https://cloud.githubusercontent.com/assets/8991915/19709986/c337d354-9b22-11e6-9cff-66bbc05d1df7.gif)

**Comment by flipdazed** [comment]
# Pretraining of Weights in DBN
Tried DBN with grater success...
![untitled](https://cloud.githubusercontent.com/assets/8991915/19826810/eb8cfcaa-9d8c-11e6-8381-3ff2530902cd.png)

**Comment by flipdazed** [comment]
closed as to general.
